src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -838,6 +838,8 @@ public async Task RespondToHandShakeAsync(CancellationToken cancellationToken =
         /// <inheritdoc/>
         public void Disconnect(string reason, Exception exception = null)
         {
+            this.logger.LogDebug("Disconnect called with reason={0} and exception={1}", reason, exception?.ToString());
```

Comment:
Hmm, maybe coalesce it to the string `"null"` so that it's apparent that no exception was passed?

Reply:
I think an empty string will be fine